spec/std/dir_spec.cr

```diff
@@ -31,9 +36,13 @@ describe "Dir" do
     end
 
     it "tests empty? on an empty directory" do
-      path = "/tmp/crystal_empty_test_#{Process.pid}/"
-      Dir.mkdir(path, 0o700)
-      Dir.empty?(path).should be_true
+      path = "#{__DIR__}/data/crystal_empty_test/"
+      begin
```

Comment:
What needs to happen is we need to provide appropriate tools to cross-compile the spec suite (i.e. never depend on the current directory and be able to tell the spec suite the location of the data and temporary dirs to use) with global helper methods. I think that's neccesary for the whole spec suite which works with files.

Reply:
Agreed. It's a different thing, though.

Reply:
Since we must change this, a helper to create/drop a tmp folder wouldn't hurt, and avoid specs to duplicate a lot of behavior.

```crystal
def with_tmpdir
  tmpdir = ENV["TMPDIR"]? || ENV["TEMP"] || ENV["TMP"] || "/tmp"
  path = File.join(tmpdir, "crystal-stdspec-#{Random.hex}")
  begin
    Dir.mkdir_p(path)
    yield path
  ensure
    FileUtils.rm_rf(path) if Dir.exists?(path)
  end
end

with_tmpdir do |path|
  Dir.mkdir(path, 0o700)
  Dir.empty?(path).should be_true
end
```

That doesn't solve the need for a `$DATADIR` environment variable, but sources and destinations are different matters.

Reply:
Thats all already covered in #5951